File: jlpt/native_ios/Sources/Data/API/AuthAPI.swift
import Foundation

final class AuthAPI {
    private let client: APIClient
    init(client: APIClient) { self.client = client }

    func register(_ request: RegisterRequestDTO) async throws -> AuthResponseDTO {
        try await client.send(Endpoint(
            path: "api/auth/register",
            method: .post,
            body: Endpoint.withJSONBody(request, encoder: client.encoder),
            requiresAuth: false
        ))
    }

    func login(_ request: LoginRequestDTO) async throws -> AuthResponseDTO {
        try await client.send(Endpoint(
            path: "api/auth/login",
            method: .post,
            body: Endpoint.withJSONBody(request, encoder: client.encoder),
            requiresAuth: false
        ))
    }
}

File: jlpt/native_android/app/src/main/java/com/jlpt/android/data/remote/api/AuthApi.kt
package com.jlpt.android.data.remote.api

import com.jlpt.android.data.remote.dto.AuthResponseDto
import com.jlpt.android.data.remote.dto.LoginRequestDto
import com.jlpt.android.data.remote.dto.RefreshRequestDto
import com.jlpt.android.data.remote.dto.RegisterRequestDto
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApi {
    @POST("api/auth/register")
    suspend fun register(@Body request: RegisterRequestDto): AuthResponseDto

    @POST("api/auth/login")
    suspend fun login(@Body request: LoginRequestDto): AuthResponseDto

    @POST("api/auth/refresh")
    suspend fun refresh(@Body request: RefreshRequestDto): AuthResponseDto
}
